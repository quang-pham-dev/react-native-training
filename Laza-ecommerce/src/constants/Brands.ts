export const BRANDS_EMPTY_RESULT = 'No Brand found';
