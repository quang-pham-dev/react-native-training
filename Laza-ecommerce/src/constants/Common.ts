//Change to http://10.0.2.2 for android and http://localhost for iOS
export const APP_BASE_URL = 'http://localhost:3000';

// Endpoints
export const ENDPOINTS = {
  AUTH: {
    LOGIN: '/auth/login',
    LOGOUT: '/auth/logout'
  },

  PRODUCT: {
    GET_PRODUCTS: '/products',
    GET_PRODUCT: '/products/:id'
  },
  BRAND: {
    GET_BRANDS: '/brands',
    GET_BRAND: '/brands/:id'
  }
};

// Auth Data key for local storage
export const AUTH_DATA = '@AuthData';

export const BOUNDARY_ERROR_MESSAGE = 'Something went wrong!';

export const SENTRY_DSN =
  'https://30f74c15a1c54630a55c03e195d86cab@o1296237.ingest.sentry.io/6522749';
