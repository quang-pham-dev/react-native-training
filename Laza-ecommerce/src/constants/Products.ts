// Constant pagination of Products
export const PRODUCT_PAGINATION = {
  PAGE: 1,
  PRODUCT_LIMIT: 6
};

// Constant messages empty result
export const PRODUCTS_EMPTY_RESULT = 'No products found';
