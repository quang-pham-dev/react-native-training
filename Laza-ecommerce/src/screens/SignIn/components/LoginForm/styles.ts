import { Platform, StyleSheet } from 'react-native';

// Themes
import Colors from 'themes/Colors';
import Fonts from 'themes/Fonts';
import Metrics from 'themes/Metrics';

export const styles = StyleSheet.create({
  main: {
    flex: 2,
    alignItems: 'center',
    justifyContent: 'center',
    paddingHorizontal: Metrics.padding.lg,
  },

  inputTitle: {
    alignSelf: 'flex-start',
    marginTop: Metrics.margin.lg,
  },
  input: {
    width: '100%',
    paddingVertical: Metrics.padding.xm,
    borderBottomWidth: 1,
    borderBottomColor: Colors.borderGray,
  },

  errorMessage: {
    alignSelf: 'flex-start',
    fontSize: Fonts.size.small,
    color: Colors.textRed,
  },

  forgotPasswordWrap: {
    marginTop: Metrics.margin.xxl,
    width: '100%',
    flexDirection: 'row',
    justifyContent: 'flex-end',
    marginBottom: Metrics.margin.lg,
  },
  forgotPassword: {
    fontFamily: 'Inter_400Regular',
    fontSize: Fonts.size.normal,
    lineHeight: Fonts.lineHeight.base,
    color: Colors.textRed,
  },

  rememberMeWrap: {
    width: '100%',
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
    marginTop: Metrics.margin.lg,
  },
  rememberMeText: {
    fontFamily: Fonts.fontFamily.Inter_500Medium,
    fontSize: Fonts.size.small,
    lineHeight: Fonts.lineHeight.xs,
    color: Colors.textBlack,
  },

  rememberMeSwitch: {
    transform:
      Platform.OS === 'android'
        ? [{ scaleX: 0.9 }, { scaleY: 0.9 }]
        : [{ scaleX: 0.6 }, { scaleY: 0.6 }],
  },

  footer: {
    flex: 1,
    justifyContent: 'flex-end',
  },

  footerTextWrapper: {
    width: '100%',
    flexDirection: 'row',
    flexWrap: 'wrap',
    paddingHorizontal: Metrics.margin.lg,
    marginBottom: Metrics.margin.xl,
  },
  condition: {
    textAlign: 'center',
    fontFamily: Fonts.fontFamily.Inter_400Regular,
    color: Colors.textGray,
    fontSize: Fonts.size.small,
    lineHeight: Fonts.lineHeight.sm,
  },
  Term: {
    color: Colors.textBlack,
    fontFamily: Fonts.fontFamily.Inter_500Medium,
  },

  bottomButton: {
    width: '100%',
    height: 75,
    paddingTop: Metrics.padding.xm,
    alignItems: 'center',
  },
  loginButton: {
    backgroundColor: Colors.primaryColor,
  },
  textBottomButton: {
    fontSize: Fonts.size.default,
    fontFamily: Fonts.fontFamily.Inter_500Medium,
    color: Colors.white,
    lineHeight: Fonts.lineHeight.sm,
  },
});

export default styles;
