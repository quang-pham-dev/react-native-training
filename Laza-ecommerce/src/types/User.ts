export interface User {
  id: string;
  avatar: string;
  username?: string;
  password: string;
  gender: string;
  email: string;
  phone: string;
  address: string;
  role: string;
}

export interface UserResponse {
  data: {
    access_token: string;
    user: User;
  };
}

export interface UserSignIn {
  username: string;
  password: string;
}
