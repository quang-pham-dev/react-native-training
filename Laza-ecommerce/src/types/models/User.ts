export interface IUser {
  id: string;
  avatar: string;
  username: string;
  password: string;
  gender: string;
  email: string;
  phone: string;
  address: string;
  role: string;
}

export interface IUserResponse {
  access_token: string;
  user: IUser;
}

export interface ILoginCredentials {
  username: string;
  password: string;
}
