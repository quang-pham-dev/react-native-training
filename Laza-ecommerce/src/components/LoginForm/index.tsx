import React, { memo, useContext, useState } from 'react';
import { Alert, Switch, Text, View } from 'react-native';
import isEqual from 'react-fast-compare';

// LIB
import { useForm, Controller, SubmitHandler } from 'react-hook-form';
import * as Yup from 'yup';
import { yupResolver } from '@hookform/resolvers/yup/dist/yup';

// Components
import Button from 'components/Button';
import TextInput from 'components/TextInput';
import LoadingIndicator from 'components/LoadingIndicator';

// Context
import { AppContext } from 'context/AppContext';
import { SIGN_IN_SUCCESS, SIGN_IN_FAILED, SIGN_IN } from 'context/actions/auth';

// API
import { authService } from 'api';

// Constants
import { AUTH_DATA } from 'constants/Common';

// Utils
import { set } from 'utils/localStorage';

// Types
import { IUserSignIn } from 'types/models/User';
import { LOADING_SIZE } from 'types/common/Enums';

// Styles
import { styles } from './styles';

// initial values
const loginFormInit: IUserSignIn = {
  username: '',
  password: '',
};

// validation schema
const loginSchema = Yup.object({
  username: Yup.string().required('Username is required'),
  password: Yup.string()
    .required('Password is required')
    .min(6, 'Password must be at least 6 characters'),
}).required();

const LoginForm = () => {
  const [isEnabled, setIsEnabled] = useState(true);
  const toggleSwitch = () => setIsEnabled(previousState => !previousState);
  const { authState, authDispatch } = useContext(AppContext);

  const {
    control,
    reset,
    handleSubmit,
    formState: { errors },
  } = useForm<IUserSignIn>({
    mode: 'onChange',
    defaultValues: loginFormInit,
    resolver: yupResolver(loginSchema),
  });

  // handle action call api SignIn when user press Login button
  const handOnSubmit: SubmitHandler<IUserSignIn> = async (loginInfo: IUserSignIn) => {
    authDispatch({ type: SIGN_IN });
    const { username, password } = loginInfo;
    try {
      const response = await authService.signIn(username, password);
      const data = JSON.stringify(response.data);
      // set auth data info to local storage
      await set(AUTH_DATA, data);
      const { user } = response.data;
      const { access_token } = response.data;
      if (user && access_token) {
        authDispatch({
          type: SIGN_IN_SUCCESS,
          payload: {
            user,
            access_token,
          },
        });
      }
    } catch (error) {
      authDispatch({ type: SIGN_IN_FAILED, payload: error });
      Alert.alert('Error', error.message);
      // reset form
      reset(loginFormInit);
    }
  };

  return (
    <>
      {authState?.isProcessing && <LoadingIndicator size={LOADING_SIZE.LARGE} />}
      <View style={styles.main}>
        {/* Username */}
        <Controller
          control={control}
          rules={{
            required: true,
          }}
          render={({ field: { onChange, value } }) => (
            <TextInput
              onChangeText={(value: string) => onChange(value)}
              value={value}
              textInputStyles={styles.input}
              label='Username'
              labelStyle={styles.inputTitle}
              placeholder='Enter your username'
              testID='usernameInput'
            />
          )}
          name='username'
        />
        {errors.username && (
          <Text testID='usernameInputError' accessibilityRole='text' style={styles.errorMessage}>
            {errors.username?.message}
          </Text>
        )}

        {/* Password */}
        <Controller
          control={control}
          rules={{
            required: true,
          }}
          render={({ field: { onChange, value } }) => (
            <TextInput
              onChangeText={(value: string) => onChange(value)}
              value={value}
              textInputStyles={styles.input}
              label='Password'
              labelStyle={styles.inputTitle}
              secureTextEntry
              placeholder='Enter your password'
              testID='passwordInput'
            />
          )}
          name='password'
        />
        {errors.password && (
          <Text testID='passwordInputError' style={styles.errorMessage}>
            {errors.password?.message}
          </Text>
        )}

        <View style={styles.forgotPasswordWrap}>
          <Text style={styles.forgotPassword}>Forgot password?</Text>
        </View>
        <View style={styles.rememberMeWrap}>
          <Text style={styles.rememberMeText}>Remember me</Text>
          <Switch
            style={styles.rememberMeSwitch}
            trackColor={{ false: '#767577', true: '#34C759' }}
            thumbColor={isEnabled ? 'white' : '#f4f3f4'}
            ios_backgroundColor='#3e3e3e'
            onValueChange={toggleSwitch}
            value={isEnabled}
          />
        </View>
      </View>
      <View style={styles.footer}>
        <View style={styles.footerTextWrapper}>
          <Text style={styles.condition}>
            By connecting your account confirm that you agree with our
            <Text style={styles.Term}> Term and Condition</Text>
          </Text>
        </View>

        <Button
          testID='loginButton'
          text='Login'
          buttonStyles={[styles.bottomButton, styles.loginButton]}
          textStyles={[styles.textBottomButton]}
          onPress={handleSubmit(handOnSubmit)}
        />
      </View>
    </>
  );
};

export default memo(LoginForm, isEqual);
