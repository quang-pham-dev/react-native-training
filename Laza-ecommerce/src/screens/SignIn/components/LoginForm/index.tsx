import React, { memo, useState } from 'react';
import { Switch, Text, View } from 'react-native';

// LIB
import isEqual from 'react-fast-compare';
import { useForm, Controller, SubmitHandler } from 'react-hook-form';
import * as Yup from 'yup';
import { yupResolver } from '@hookform/resolvers/yup/dist/yup';

// Components
import Button from 'components/Button';
import TextInput from 'components/TextInput';

// Types
import { ILoginCredentials } from 'types/models/User';

// Styles
import { styles } from './styles';
import Colors from 'themes/Colors';

// initial values
const loginFormInit: ILoginCredentials = {
  username: '',
  password: '',
};

// validation schema
const loginSchema = Yup.object({
  username: Yup.string().required('Username is required'),
  password: Yup.string()
    .required('Password is required')
    .min(6, 'Password must be at least 6 characters'),
}).required();

interface ILoginFormProps {
  onSubmit: (data: ILoginCredentials) => void;
}

const LoginForm = ({ onSubmit }: ILoginFormProps) => {
  const [isEnabled, setIsEnabled] = useState(true);
  const toggleSwitch = () => setIsEnabled(previousState => !previousState);
  const {
    control,
    handleSubmit,
    formState: { errors },
  } = useForm<ILoginCredentials>({
    mode: 'onChange',
    defaultValues: loginFormInit,
    resolver: yupResolver(loginSchema),
  });

  const onSubmitHandler: SubmitHandler<ILoginCredentials> = async (
    loginInfo: ILoginCredentials,
  ) => {
    onSubmit(loginInfo);
  };

  return (
    <>
      <View style={styles.main}>
        {/* Username */}
        <Controller
          control={control}
          rules={{
            required: true,
          }}
          render={({ field: { onChange, value } }) => (
            <TextInput
              onChangeText={(value: string) => onChange(value)}
              value={value}
              textInputStyles={styles.input}
              label='Username'
              labelStyle={styles.inputTitle}
              placeholder='Enter your username'
              testID='usernameInput'
            />
          )}
          name='username'
        />
        {errors.username && (
          <Text testID='usernameInputError' accessibilityRole='text' style={styles.errorMessage}>
            {errors.username?.message}
          </Text>
        )}

        {/* Password */}
        <Controller
          control={control}
          rules={{
            required: true,
          }}
          render={({ field: { onChange, value } }) => (
            <TextInput
              onChangeText={(value: string) => onChange(value)}
              value={value}
              textInputStyles={styles.input}
              label='Password'
              labelStyle={styles.inputTitle}
              secureTextEntry
              placeholder='Enter your password'
              testID='passwordInput'
            />
          )}
          name='password'
        />
        {errors?.password && (
          <Text testID='passwordInputError' style={styles.errorMessage}>
            {errors?.password?.message}
          </Text>
        )}

        <View style={styles.forgotPasswordWrap}>
          <Text style={styles.forgotPassword}>Forgot password?</Text>
        </View>
        <View style={styles.rememberMeWrap}>
          <Text style={styles.rememberMeText}>Remember me</Text>
          <Switch
            style={styles.rememberMeSwitch}
            trackColor={{ false: Colors.lightGray, true: Colors.lightGreen }}
            thumbColor={isEnabled ? Colors.white : Colors.lightGray}
            ios_backgroundColor={Colors.lightGray}
            onValueChange={toggleSwitch}
            value={isEnabled}
          />
        </View>
      </View>
      <View style={styles.footer}>
        <View style={styles.footerTextWrapper}>
          <Text style={styles.condition}>
            By connecting your account confirm that you agree with our
            <Text style={styles.Term}> Term and Condition</Text>
          </Text>
        </View>

        <Button
          testID='loginButton'
          text='Login'
          buttonStyles={[styles.bottomButton, styles.loginButton]}
          textStyles={[styles.textBottomButton]}
          onPress={handleSubmit(onSubmitHandler)}
        />
      </View>
    </>
  );
};

export default memo(LoginForm, isEqual);
