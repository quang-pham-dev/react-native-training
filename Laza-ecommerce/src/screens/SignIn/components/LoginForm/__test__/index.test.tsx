import React from 'react';

// LIBS
import { act, fireEvent, render } from '@testing-library/react-native';
import AsyncStorage from '@react-native-async-storage/async-storage';
import renderer from 'react-test-renderer';

// Components
import LoginForm from 'screens/SignIn/components/LoginForm';

// constants
import { AUTH_DATA } from 'constants/Common';

// Types
import { ILoginCredentials } from 'types/models/User';

const mockLogin = jest.fn((data: ILoginCredentials) => {
  return Promise.resolve({ data });
});

describe('Login Form', () => {
  let tree: any;
  beforeEach(() => {
    tree = render(<LoginForm onSubmit={mockLogin} />);
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  test('should render correctly', () => {
    expect(tree).toMatchSnapshot();
  });

  test('should render forgot password text', () => {
    const { getByText } = tree;
    expect(getByText('Forgot password?')).toBeTruthy();
  });

  test('should render remember me text', () => {
    const { getByText } = tree;
    expect(getByText('Remember me')).toBeTruthy();
  });

  test('should not render error messages when input values are valid', async () => {
    const { queryAllByText, getByPlaceholderText } = tree;
    const usernameValue = 'Quangpham';
    const passwordValue = '12345';

    await act(async () =>
      fireEvent.changeText(getByPlaceholderText('Enter your username'), usernameValue),
    );
    await act(async () =>
      fireEvent.changeText(getByPlaceholderText('Enter your password'), passwordValue),
    );

    const errorMessage = queryAllByText('Password must be at least 6 characters');

    expect(errorMessage).not.toBeNull();
  });

  //check with mock async storage value
  test('checks if Async Storage is used', async () => {
    await AsyncStorage.getItem(AUTH_DATA);

    expect(AsyncStorage.getItem).toBeCalledWith(AUTH_DATA);
  });
});
