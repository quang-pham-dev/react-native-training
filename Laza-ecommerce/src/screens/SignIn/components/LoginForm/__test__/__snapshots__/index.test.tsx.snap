// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Form should render correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 3,
        "justifyContent": "center",
        "paddingHorizontal": 20,
      }
    }
  >
    <View
      style={
        Object {
          "width": "100%",
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "alignSelf": "flex-start",
              "color": "#8F959E",
              "fontFamily": "Inter_400Regular",
              "fontSize": 13,
              "lineHeight": 14,
              "marginTop": 20,
            },
            undefined,
          ]
        }
      >
        Username
      </Text>
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "center",
            "width": "100%",
          }
        }
      >
        <TextInput
          allowFontScaling={true}
          onChangeText={[Function]}
          placeholder="Enter your username"
          rejectResponderTermination={true}
          secureTextEntry={false}
          style={
            Array [
              Object {
                "borderBottomColor": "#E7E8EA",
                "borderBottomWidth": 1,
                "fontFamily": "Inter_500Medium",
                "fontSize": 15,
                "lineHeight": 16,
                "paddingRight": 25,
                "paddingVertical": 15,
                "width": "100%",
              },
              undefined,
            ]
          }
          underlineColorAndroid="transparent"
          value=""
        />
        <View
          accessible={true}
          focusable={false}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
              "position": "absolute",
              "right": 0,
            }
          }
        >
          <Text />
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "width": "100%",
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "alignSelf": "flex-start",
              "color": "#8F959E",
              "fontFamily": "Inter_400Regular",
              "fontSize": 13,
              "lineHeight": 14,
              "marginTop": 20,
            },
            undefined,
          ]
        }
      >
        Password
      </Text>
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "center",
            "width": "100%",
          }
        }
      >
        <TextInput
          allowFontScaling={true}
          onChangeText={[Function]}
          placeholder="Enter your password"
          rejectResponderTermination={true}
          secureTextEntry={true}
          style={
            Array [
              Object {
                "borderBottomColor": "#E7E8EA",
                "borderBottomWidth": 1,
                "fontFamily": "Inter_500Medium",
                "fontSize": 15,
                "lineHeight": 16,
                "paddingRight": 25,
                "paddingVertical": 15,
                "width": "100%",
              },
              undefined,
            ]
          }
          underlineColorAndroid="transparent"
          value=""
        />
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
              "position": "absolute",
              "right": 0,
            }
          }
        >
          <Text />
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "flex-end",
          "marginBottom": 20,
          "marginTop": 30,
          "width": "100%",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#EA4335",
            "fontFamily": "Inter_400Regular",
            "fontSize": 15,
            "lineHeight": 16,
          }
        }
      >
        Forgot password?
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginTop": 20,
          "width": "100%",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#1D1E20",
            "fontFamily": "Inter_500Medium",
            "fontSize": 13,
            "lineHeight": 14,
          }
        }
      >
        Remember me
      </Text>
      <View>
        <RCTSwitch
          accessibilityRole="switch"
          onChange={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          onTintColor="#34C759"
          style={
            Array [
              Object {
                "height": 31,
                "width": 51,
              },
              Array [
                Object {
                  "transform": Array [
                    Object {
                      "scaleX": 0.6,
                    },
                    Object {
                      "scaleY": 0.6,
                    },
                  ],
                },
                Object {
                  "backgroundColor": "#F2F2F2",
                  "borderRadius": 16,
                },
              ],
            ]
          }
          thumbTintColor="#fff"
          tintColor="#F2F2F2"
          value={true}
        />
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "flex": 1,
        "justifyContent": "flex-end",
      }
    }
  >
    <View
      style={
        Object {
          "flexDirection": "row",
          "flexWrap": "wrap",
          "marginBottom": 25,
          "paddingHorizontal": 20,
          "width": "100%",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#8F959E",
            "fontFamily": "Inter_400Regular",
            "fontSize": 13,
            "lineHeight": 18,
            "textAlign": "center",
          }
        }
      >
        By connecting your account confirm that you agree with our
        <Text
          style={
            Object {
              "color": "#1D1E20",
              "fontFamily": "Inter_500Medium",
            }
          }
        >
           Term and Condition
        </Text>
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#3D93F8",
          "height": 75,
          "opacity": 1,
          "paddingTop": 15,
          "width": "100%",
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#fff",
              "fontFamily": "Inter_500Medium",
              "fontSize": 17,
              "lineHeight": 18,
              "textAlign": "center",
            },
            undefined,
          ]
        }
      >
        Login
      </Text>
    </View>
  </View>
</View>
`;
